src/databases/PLOT3D/vtkPLOT3DReader.C

```diff
@@ -790,20 +823,77 @@ vtkPLOT3DReader::ReadGrid(FILE *xyzFp)
   this->SkipByteCount(xyzFp);
   int d = this->Internal->NumberOfDimensions;
   if (this->ReadVector(xyzFp, this->NumberOfPoints, d, pointArray) == 0)
-    {
+  {
     vtkErrorMacro("Encountered premature end-of-file while reading "
                   "the geometry file (or the file is corrupt).");
     return VTK_ERROR;
-    }
+  }
 
-  // This is where we would read IBlanking information, but VisIt currently 
-  //  doesn't do anything with it
-  // 
-#if 0
+  // Read IBlanking data
   if (this->Internal->IBlanking)
+  {
+    vtkIntArray* iblank = vtkIntArray::New();
+    iblank->SetName("avtGhostNodes");
+    iblank->SetNumberOfTuples(this->NumberOfPoints);
+
+    // Attempt to read the iblank data from the plot3d file. If the read failes,
+    // then the function will return 0 and we will error out. If the read passes,
+    // it will return the count that indicates how many numbers it read.
+    if (this->ReadIBlank(xyzFp, this->NumberOfPoints, iblank) == 0)
+    {
+      vtkErrorMacro("Encountered premature end-of-file while reading "
+                    "the iblanking data from the xyz file.");
+      return VTK_ERROR;
+    }    
+
+    // IBlank data is 0 if it should be blanked and 1 if it should not be
+    // blanked, which is a little backwards from how I think of it, but okay.
+    // So where ib is 0 we use NODE_NOT_APPLICABLE_TO_PROBLEM and where ib is
+    // 1 we use DUPLICATED_NODE.
+    int* ib = iblank->GetPointer(0);
+    for (int i = 0; i < this->NumberOfPoints; ++i)
     {
+      if (ib[i] == 0)
+      {
+        ib[i] = avtGhostNodeTypes::NODE_NOT_APPLICABLE_TO_PROBLEM;
+      }
+      else if (ib[i] == 1)
+      {
+        ib[i] = avtGhostNodeTypes::DUPLICATED_NODE;
```

Comment:
We're overwritting VTK's `ib` data with VisIt ghost node type data here, right?

In the else block here, it doesn't make sense IMHO to assign VisIt's `avtGhostNodeType::DUPLICATED_NODE`. Instead, I think it should just clear `ib[i]` to zero `0` here as in no special ghost type is assigned for node `i`.

Reply:
> 
> 
> We're overwritting VTK's `ib` data with VisIt ghost node type data here, right?
> 
> In the else block here, it doesn't make sense IMHO to assign VisIt's `avtGhostNodeType::DUPLICATED_NODE`. Instead, I think it should just clear `ib[i]` to zero `0` here as in no special ghost type is assigned for node `i`.

@markcmiller86 I was confused by the fact that there isn't a value in avtGhostTypes to represent 'not-a-ghost' either for node or zone ghosts. Setting a node's ghost type to 0 is the same as setting it to DUPLICATED_NODE, codewise, is it not?

Reply:
Argh! Indeed `DUPLICATED_NODE` has value zero and that means `avtGhostTypes` left shifts 0 bits to assign the value for `ib[i]`.

I think `avtGhostTypes` is poorly designed in this regard then because it means it is not possible to have a node or a zone that has no *ghost* designation. Every node or zone is some kind of ghost. In particular, a zero'd ghost node array then means that all nodes are *duplicated*. Likewise for a ghost zone array. Does that make conceptual sense to anyone? It sure doesn't to me. It seems misleading.

`avtGhostTypes` desgin aside, using the `avtGhostType` symbol `DUPLICATED_NODE` in this code block in `vtkPLOT3DReader.C` IMHO repeats that confusion. Most readers could read `0` there and probably conclude *oh, ok, that branch is for nodes that are not ghosts* whereas now, readers are going to conclude that branch is for a different kind of ghost. In particular, a duplicated node.

You can only conclude, as you did, that it turns out the designation `DUPLICATED_NODE` has no effect if they go read the implementation of `avtGhostData` and discover

https://github.com/visit-dav/visit/blob/dfc29a1457ea9fc330befa7fc0b21ba1705946f7/src/avt/Pipeline/Data/avtGhostData.h#L109-L112

which returns `true` only for non-zero data.

Reply:
Actually, I am totally reading that ghosting logic in `avtGhostData.h` incorrectly! And, we appear to be using it incorrectly here and elsewhere too. 

`DUPLICATED_NODE` is a real ghost type and using it *should* result in a value of `0x01` in the ghost data value. That is because it is OR'ing in a `1` left shifted by `0` positions.

So, I rescind my comments regarding design of `avtGhostType` and `avtGhostData` and I'll amplify my point, don't use `DUPLICATED_NODE` there. Instead, assign 0.

Reply:
Oh, and we should never be *assigning* values of `avtGhostType::<SOME GHOST ENUM SYMBOL>`. We should be using the `AddGhostNodeType()` and `AddGhostZoneType()` functions there which handle the bit-shifting and or'ing.

Reply:
I agree with Mark, you should assign 0 if not a ghost node. If it is a ghost node, AddGhostNodeType() should be used. I also agree that this is very confusing.

Reply:
Got it. I wasn't aware of the bit-shifting properties in ghost nodes/zones. I just saw the enumeration and hoped that duplicate node would be treated like a non-ghost-node since it is zero.

Reply:
@brugger1 @markcmiller86 I've updated they way ghost nodes/zones are added. Please take a look and let me know if this is correct. I'm still seeing some weird behavior in the tests, which I'm trying to debug, and this may be the source of it if I've done this incorrectly.

Reply:
I think gnp will be uninitialized memory when it is created and since AddGhostNodeType() adds the bit shifted quantity to gnp[i], the result is undefined. I think you should zero gnp[i] before calling AddGhostNodeType(gnp[i],...).